openai/openai_test.go

```diff
@@ -195,7 +195,86 @@ func TestChatMiddleware(t *testing.T) {
 				Stream: &False,
 			},
 		},
-
+		{
+			name: "chat handler with streaming tools",
```

Comment:
Quick test to make sure middleware is catching streamed tools (nothing for this PR but just general tracking regressions)